### ISSUE TITLE
GridAggregator result not match the input.

### ISSUE THREAD

**Issue by Hsuxu** [fepegar/torchio#813]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

When use a large `patch_size` and a non-negative `patch_overlab` in `GridSampler`, the aggregate result is not match with the input.

### Code for reproduction

```python
import torch
import torch.nn as nn
import torchio as tio
patch_overlap = 4  # or just 4
patch_size = 102

img = torch.zeros((1, 320, 320, 320))
bbox = [100,100,100,120,120,120]

img[:,bbox[0]:bbox[3],bbox[1]:bbox[4],bbox[2]:bbox[5]]=1

subject = tio.Subject(image=tio.ScalarImage(tensor=img))
# subject
# Colin27(Keys: ('t1', 'head', 'brain'); images: 3)
grid_sampler = tio.inference.GridSampler(
    subject,
    patch_size,
    patch_overlap,
)
patch_loader = torch.utils.data.DataLoader(grid_sampler, batch_size=4)
aggregator = tio.inference.GridAggregator(grid_sampler)
model = nn.Identity().eval()
with torch.no_grad():
    for patches_batch in patch_loader:
        input_tensor = patches_batch['image'][tio.DATA]
        locations = patches_batch[tio.LOCATION]
        logits = model(input_tensor)
#         labels = logits.argmax(dim=tio.CHANNELS_DIMENSION, keepdim=True)
#         outputs = labels
        aggregator.add_batch(input_tensor, locations)
output_tensor = aggregator.get_output_tensor()
indices = torch.nonzero(output_tensor==1)
sp,_ = indices.min(0)
ep,_ = indices.max(0)
print(sp,ep)
```

### Actual outcome

result `tensor([ 0, 99, 99, 99]) tensor([  0, 119, 119, 119])`  for `patch_overlab = 4;patch_size = 102` 
result `tensor([ 0, 98, 98, 98]) tensor([  0, 119, 119, 119])` for `patch_overlap = 8;patch_size = 120`

### Error messages

```python-traceback
no error message
```

### Expected outcome

tensor([  0, 100, 100, 100]) tensor([  0, 119, 119, 119])

### System info

```shell
Platform:   Linux-4.4.0-142-generic-x86_64-with-debian-stretch-sid
TorchIO:    0.18.73
PyTorch:    1.8.0
SimpleITK:  2.1.1 (ITK 5.2)
NumPy:      1.19.2
Python:     3.6.13 |Anaconda, Inc.| (default, Feb 23 2021, 21:15:04)
[GCC 7.3.0]
```

**Comment by fepegar** [comment]
Hi, @Hsuxu. What do you mean by "the aggregate result is not match with the input"?

**Comment by Hsuxu** [comment]
As shown in the code,  pass a binary mask with a bounding box is positive, while the output is not same with input.
For example, modity the code like this. The input full image with binary mask which from `[100,100,100]` to `[119,119,119]` is positive.
```
patch_overlap = 32 # or just 4
patch_size = 160
img = torch.zeros((1, 320, 320, 320))
bbox = [100,100,100,120,120,120]
```

The output of aggragato where is equal to `1` is from `[92,92,92]` to `[111,111,111]` 
```
tensor([ 0, 92, 92, 92]) tensor([  0, 111, 111, 111])
```
The input and output are shown in the figure, binary mask is the input of the aggregator, red overlay is the output.
![image](https://user-images.githubusercontent.com/15630477/153986600-8b74ec0e-7930-4707-900c-e11bffe05aab.png)

**Comment by fepegar** [comment]
You're right, something seems off. I'm looking into it now.

**Comment by fepegar** [comment]
The bug is in the aggregator. The indices are not computed correctly in this function:

https://github.com/fepegar/torchio/blob/eddd02decf2f00777429452e6056fdf72674d8d3/torchio/data/inference/aggregator.py#L51-L56
